src/posthog-core.ts

```diff
@@ -764,6 +764,19 @@ export class PostHog {
             this.persistence.set_initial_referrer_info()
         }
 
+        if (
```

Comment:
As a user I would find this pretty confusing but I guess thats a different discussion.

Code-wise, this is a BadIdea™️ - If I do something like

```js
const setContext = getSetProperties() // Returns {}

posthog.capture("my-event", { $set: setContext })
```

There are 2 problems:
1. I am "upgraded" despite there being no properties to actually be set
2. The event is dropped (what) simply because that property is set.

Personally I think the solution should be that we ignore set properties and just let them through, not using them as a decision point to upgrade to person processing. We can document it as such and say `If you choose to use $set you need to make sure the config setting is turned on otherwise person profiles will not be created` or something.

Reply:
I think it's reasonable to do the empty object check for sure, I can add that, but IMO it does make sense to upgrade the user to person processing if they are trying to $set properties. That's kinda what `identified-only` is, you only pay for person processing once you use it!

It's better to fail expensive than fail data loss, as we can always refund people later but we can't recover their data.

Reply:
It's more a comment on the overall API design - this is confusing and hidden. I'd much rather we thought about being explicit about this somehow but I understand there's probably low appetite for that atm.